pkg/kv/kvserver/spanset/spanset_test.go

```diff
@@ -203,25 +204,25 @@ func TestSpanSetCheckAllowedAtTimestamps(t *testing.T) {
 	ss.AddNonMVCC(SpanReadWrite, roachpb.Span{Key: keys.RangeGCThresholdKey(1)})
 
 	var allowedRO = []struct {
```

Comment:
No tests here with `TrickySpan{EndKey: something}`. Could be good to add some. We have good coverage for it, so optional / TODO.

More generally, would be also good to de-boilerplate these tests. Also feel free to ignore / TODO.

Same below.